### DISCUSSION TITLE
DXF data & attributes

### DISCUSSION THREAD

**Discussion by f0rme** [mozman/ezdxf#788] [Q&A] [answered]
Two questions related to DXF data. 

1. Is there a way to remove application specific data from a DXF file? Sometimes the most simple drawings are 3+mb in size because they contain data that may be used by other applications. In some cases 100,000's of lines of text. 
Is there a way to remove data inserted for use by a specific application?  

2. Is there a way to add custom attributes to an entity? 
For example:

AcDbPolyline
 90
        4
 70
   129
 43
0.0
 10
0.0
 20
0.0
 10
100.0
 20
0.0
 10
100.0
 20
100.0
 10
0.0
 20
100.0
1001
SOMETEXT
1250
Group:Entrances

Note the 'SOMETEXT' attribute near the bottom of the polyline 'description'.

**Comment by mozman** [comment, marked as answer]
Discard XDATA for AppID "SOMETEXT" in all entities:

```Python
for entity in doc.entitydb.values():
    entity.discard_xdata("SOMETEXT")

```

This method discards only the XDATA for the given AppID, it's possible to delete all XDATA, but this is not recommended because also AutoCAD stores important information in the XDATA section by multiple AppIDs.

**Reply by f0rme** [reply]
Thank you. I will look at the documentation for XDATA.
